prow/plugins/projectmanager/projectmanager_test.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package projectmanager
+
+import (
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+func TestHandlePR(t *testing.T) {
+	columnID := new(int)
+	cases := []struct {
+		name                string
+		gc                  *fakegithub.FakeClient
+		projectManager      plugins.ProjectManager
+		pe                  github.PullRequestEvent
+		expectedColumnCards map[int][]github.ProjectCard
+	}{
+		{
+			name: "add pull request to project column with no columnID",
+			gc: &fakegithub.FakeClient{
+				OrgRepoIssueLabels: map[string][]github.Label{
+					"otherOrg/someRepo/1": {
+						{
+							Name: "label1",
+						},
+						{
+							Name: "label2",
+						},
+					},
+				},
+				RepoProjects: map[string][]github.Project{
+					"testOrg/testRepo": {
+						{
+							Name: "testProject",
+							ID:   1,
+						},
+					},
+				},
+				OrgProjects: map[string][]github.Project{},
+				ProjectColumns: map[int][]github.ProjectColumn{
+					1: {
+						{
+							Name: "testColumn",
+							ID:   1,
+						},
+					},
+				},
+				ColumnCardsMap: map[int][]github.ProjectCard{},
+			},
+			projectManager: plugins.ProjectManager{
+				OrgRepos: map[string]plugins.ManagedOrgRepo{
+					"testOrg/testRepo": {
+						Projects: map[string]plugins.ManagedProject{
+							"testProject": {
+								Columns: []plugins.ManagedColumn{
+									{
+										Name:   "testColumn",
+										State:  "open",
+										Org:    "otherOrg",
+										Labels: []string{"label1", "label2"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			pe: github.PullRequestEvent{
+				Action: github.PullRequestActionOpened,
+				Number: 1,
+				PullRequest: github.PullRequest{
+					ID:    2,
+					State: "open",
+				},
+				Repo: github.Repo{
+					Name: "someRepo",
+					Owner: github.User{
+						Login: "otherOrg",
+					},
+				},
+			},
+			expectedColumnCards: map[int][]github.ProjectCard{
+				1: {
+					{
+						ContentID:   2,
+						ContentType: "PullRequest",
+					},
+				},
+			},
+		},
+		{
+			name: "add pull request to project column with only columnID",
```

Comment:
This test case has more than just `columnID` specified.
The `expectedColumnCards` is also identical to the previous test case's so this test case doesn't actually appear to be validating the behavior. (I would expect a different expected column ID to be used for this test case so that we can distinguish the behavior in these two cases.)

Additionally, the value of `columnID` is 0, but the test case is still passing even though the `expectedColumnCards` indicates that a card is added to column 1. This looks like a bug that the test failed to detect. Please confirm if this is the case and update the code and test if it is.

Reply:
Modified the tests, tested with column ID's for multiple projects